py3status/modules/coin_market.py

```diff
@@ -0,0 +1,223 @@
+                break
+
+        # create '(?|&)limit'
+        if self.limit:
+            self._update_limit(None)
+
+    def _get_coin_data(self, reset=False):
+        if reset:
+            self.url = self.reset_url
+        try:
+            data = self.py3.request(self.url, timeout=self.request_timeout).json()
+        except self.py3.RequestException:
+            data = {}
+        return data
+
+    def _update_limit(self, data):
+        # we use limit if it exists. otherwise, we stretch the limit
+        # large enough to obtain (all) self.markets + some padding
+        self.url = self.url + ('&' if self.convert else '?')
+        if self.limit:
+            limit = self.limit
+        else:
+            limit = 0
+            for market_id in self.markets:
+                index = next((i for (i, d) in enumerate(
+                    data) if d['symbol'] == market_id), -1)
+                if index >= limit:
+                    limit = index
+                    limit += 5  # padding
+
+        self.url += 'limit=%s' % limit
+
+    def _strip_data(self, data):
+        # if self.limit, we don't strip. otherwise, we strip 1000+ coins
+        # down to %s coins by removing everything not in self.markets.
+        new_data = []
+        if self.limit:
+            new_data = data
+        else:
+            for symbol in self.markets:
+                for market in data:
+                    if symbol == market['symbol']:
+                        new_data.append(market)
+                        break
+
+        return new_data
+
+    def _organize_data(self, data):
+        # compare len(self.markets) and len(stripped(900+ coins))
+        new_data = self._strip_data(data)
+        is_equal = len(new_data) == len(self.markets)
+
+        # first_use bad? the user entered bad markets. stop here (error).
+        # otherwise, make a limit for first time on 1000+ coins.
+        if self.first_use:
+            self.first_use = False
+            if not is_equal:
+                self.py3.error('bad markets')
+            else:
+                self._update_limit(data)
+        elif not is_equal:
+            # post first_use bad? the markets fell out of the limit + padding.
+            # reset the url to get 1000+ coins again so we can strip, compare,
+            # make new limit + padding for next loop, but we'll use that new
+            # data. otherwise, we would keep going with the stripped data.
+            new_data = self._get_coin_data(reset=True)
+            new_data = self._strip_data(new_data)
+            self._update_limit(new_data)
+
+        return new_data
+
+    def _manipulate_data(self, data):
+        # we mess with raw data to get the results we want. we're only
+        # adding colors + symbols to all percent placeholders.
+        new_data = []
+        for market in data:
+            temporary = {}
+            for k, v in market.items():
+                if 'percent_change' in k and v:
```

Comment:
I think paranoid. I want to leave it as-is and wait until the shit hits the fan first.